<launch>
	<arg name="camera" default="camera" />
	<node pkg="fovis_ros" type="fovis_mono_depth_odometer" name="visual_odometer" >
		<remap from="/camera/rgb/image_rect" to="$(arg camera)/rgb/image_rect_color" />
	    <remap from="/camera/rgb/camera_info" to="$(arg camera)/rgb/camera_info" />
	    <remap from="/camera/depth_registered/camera_info" to="$(arg camera)/depth_registered/camera_info" />
	    <remap from="/camera/depth_registered/image_rect" to="$(arg camera)/depth_registered/hw_registered/image_rect_raw" />      
	    <param name="approximate_sync" type="bool" value="True" />	     
	    <param name="base_link_frame_id" value="camera_link" />	   
	    <!--<param name="odom_frame_id" value="odom" />-->
 	    <param name="publish_tf" value="false" />
	</node>

	<node pkg="carl_navigation" type="odom_covariance_converter" name="odom_convert">
		<remap from="/odom_convert/odom_in" to="odom" />
		<remap from="/odom_convert/odom_out" to="covariance_odom" />
		<param name="cov_x" type="double" value="1e-3" />
		<param name="cov_y" type="double" value="1e-3" />
		<param name="cov_z" type="double" value="1e-3" />
		<param name="cov_rx" type="double" value="1e9" />
		<param name="cov_ry" type="double" value="1e9" />
		<param name="cov_rz" type="double" value="1e6" />
	</node>	

	<node pkg="carl_navigation" type="odom_covariance_converter" name="vo_convert">
		<remap from="/vo_convert/odom_in" to="/visual_odometer/odometry" />
		<remap from="/vo_convert/odom_out" to="/covariance_vo" />
		<param name="cov_x" type="double" value="1e6" />
		<param name="cov_y" type="double" value="1e6" />
		<param name="cov_z" type="double" value="1e6" />
		<param name="cov_rx" type="double" value="1e9" />
		<param name="cov_ry" type="double" value="1e9" />
		<param name="cov_rz" type="double" value="1e3" />
	</node>	

	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
	    <param name="output_frame" value="odom_combined"/>
	    <param name="freq" value="60.0"/>
	    <param name="sensor_timeout" value="1.0"/>
	    <param name="odom_used" value="true"/>
	    <param name="imu_used" value="false"/>
	    <param name="vo_used" value="true"/>
	    <param name="debug" value="false"/>
  		<param name="self_diagnose" value="false"/>
		<remap from="/vo" to="/covariance_vo"/>
		<remap from="/odom" to="/covariance_odom" />
	</node>
</launch>
